src/components/Settings/ImageUpload.tsx

```diff
@@ -41,37 +48,65 @@ const ImageUpload = ({
         ref={imageRef}
         style={{ display: 'none' }}
         onChange={handleChange}
+        disabled={isLoading}
       />
-      <Image
-        {...rest}
-        cursor="pointer"
-        overflow="hidden"
-        bgColor="rgba(130, 130, 130, 0.49)"
-        position="relative"
-        _hover={{
-          _after: {
-            content: '""',
-            position: 'absolute',
-            top: '0',
-            left: '0',
-            bottom: '0',
-            right: '0',
-            bg: 'rgba(0, 0, 0, 0.5)',
-          },
-          _before: {
-            content: 'url(/images/edit_icon.svg)',
-            position: 'absolute',
-            left: '50%',
-            top: '50%',
-            transform: 'translate(-50%, -50%)',
-            zIndex: 999,
-          },
-        }}
-        src={defaultUserImage}
-        onClick={() => imageRef.current?.click()}
-      />
+      <Box pos="relative" w="max-content">
+        {isLoading && (
+          <Box
+            pos="absolute"
+            top="50%"
+            left="50%"
+            transform="translate(-50%, -50%)"
+            zIndex={9}
+            bgColor="pageBg"
+            p={2}
+            borderRadius="50%"
+          >
```

Comment:
I think this could just use `Center`
```suggestion
      <Center w="max-content">
        {isLoading && (
          <Box
            zIndex={9}
            bgColor="pageBg"
            p={2}
            borderRadius="50%"
          >

Reply:
Ohh thanks ! i will test it out with center

Reply:
@anubra266 nope did not work, the loader went side to the image instead of being on top.